src/signature_accumulator.rs

```diff
@@ -210,7 +210,7 @@ mod tests {
                         msg_and_sigs.signed_msg.routing_message(),
                         returned_msg.routing_message()
                     );
-                    unwrap!(returned_msg.check_integrity(1000));
+                    unwrap!(returned_msg.check_integrity(env.num_nodes()));
```

Comment:
Related, to comment on has_enough_sigs,
Maybe remove that parameter so your compiler can tell you all the place you won't need it, and it seem we should not need to provide it.

Reply:
if there is no concern as mentioned in the above https://github.com/maidsafe/routing/pull/1707#discussion_r300686385
Then, I will try to remove this parameter.